build.sbt

```diff
@@ -9,11 +9,12 @@ scalaVersion := "2.12.10"
 crossScalaVersions := Seq("2.12.10", "2.13.1")
 
 libraryDependencies ++= Seq(
-  "com.typesafe"          %  "config"         % "1.3.3",
-  "org.specs2"           %%  "specs2-core"    % "4.6.0" % "test",
+  "com.typesafe"           %  "config"                  % "1.3.3",
+  "org.specs2"             %%  "specs2-core"            % "4.6.0" % "test",
   "org.scala-lang.modules" %% "scala-collection-compat" % "2.1.4",
-  "org.json4s"           %%  "json4s-native"  % "3.6.7",
-  "com.google.code.gson"  %  "gson"           % "2.8.0"
+  "org.json4s"             %%  "json4s-native"          % "3.6.7",
+  "com.google.code.gson"   %  "gson"                    % "2.8.0",
+  "org.scala-graph"        %% "graph-core"              % "1.13.1"
```

Comment:
Didn't think an extra library would be needed for the task of finding cycles....  but perhaps it's a very lightweight library anyway.